cmake/scripts/generate_config.cmake

```diff
@@ -34,7 +34,7 @@ endif()
 
 set(ONEDPL_ROOT "${CMAKE_CURRENT_LIST_DIR}/../..")
 
-file(READ ${ONEDPL_ROOT}/include/oneapi/dpl/pstl/onedpl_config.h _onedpl_version_info LIMIT 1024)
+file(READ ${ONEDPL_ROOT}/include/oneapi/dpl/internal/version_impl.h _onedpl_version_info LIMIT 1024)
```

Comment:
It seems to me this constant '1024' similar to some tricks that rely on the actual content of the `include/oneapi/dpl/internal/version_impl.h` file and it may breaks if this file will edited later.

Reply:
https://cmake.org/cmake/help/latest/command/file.html
```
file(READ <filename> <variable> [OFFSET <offset>] [LIMIT <max-in>] [HEX])
Read content from a file called <filename> and store it in a <variable>.
Optionally start from the given <offset> and read at most <max-in> bytes.
```

Currently the needed lines fit into the first 1024 bytes. The new file is ~10x smaller than `onedpl_config.h`, currently ~1.4K. So maybe the limit has now less value, but I would let others to decide.

Reply:
@dmitriy-sobolev, @danhoeflinger what do you think about this?

Reply:
I think it is fine as is.  At most we could add a comment in version_impl.h saying it should not grow beyond 1024 bytes without changes in this file, but I think were probably OK without it.

Reply:
I have no strong opinion on that. You can follow Dan's suggestion, or remove LIMIT. I do not expect that the file will significantly inflate (I suppose LIMIT was added to speed up CMake configuration).

Reply:
OK. So let's save as is.

Reply:
Just for note: we reads this file in two places:
- in the `scripts/generate_config.cmake` file;
- in the `CMakeLists.txt` file.
I mean if we decide to remove `LIMIT` usage, required to remove it in both places.

Reply:
I moved the library version macros to the top of the file.